bot/modules/study/time_tracker.py

```diff
@@ -105,5 +108,87 @@ async def launch_study_tracker(client):
     await admin.untracked_channels.launch_task(client)
     asyncio.create_task(_study_tracker())
 
+# only whitelisted non-bot member
+def member_joined_voice_channel(user, member, joinedchannel):
```

Comment:
A couple of notes on this system. **You don't need to resolve them** because the time tracker is deprecated to be replaced by the session tracker, but just for your information.
* The session log tracks start and end coins, but doesn't account for the coins earnt during the session. E.g. if they complete a TODO during the session it will be logged as their session coins.
* This doesn't account for untracked channels
* This doesn't account for switching channels, only explicitly joining and leaving them (perhaps only relevant in conjunction with untracked channels).
* The system needs a "load" task. When it goes offline (or Discord goes offline), users will leave and join sessions arbitrarily.. which will mean scrambled data after loading.
* The poll loop won't pick up joins before the client is ready. The event should be added after the load task has handled the new state.
* Use a batch update to write multiple pieces of data, each write is a database call.
* Event logging every single channel join is a *lot* of logging, which means a lot of extra events on a system which is already skirting the global ratelimits. Consider the accountability system across thousands of servers as dozens of users join *each* accountability channel at the same time. We would be temp banned for ratelimits on the first hour! An alternative is to batch log messages and send them in groups in single messages, but it's not worth logging joins that way.

Overall well implemented though, nicely done!